lib/IRGen/GenMeta.cpp

```diff
@@ -2058,6 +2058,13 @@ namespace {
         return emitFromValueWitnessTable(IGF.IGM.Context.TheEmptyTupleType);
       }
       case MetatypeRepresentation::Thick:
+        if (isa<ExistentialMetatypeType>(type)) {
+            return emitFromTypeMetadata(type);
+        } else {
+          // Thick metatypes look like pointers with spare bits.
+          return emitFromValueWitnessTable(
+                     CanMetatypeType::get(IGF.IGM.Context.TheNativeObjectType));
```

Comment:
Should this just fall through, since the ObjC case below is the same?

Reply:
I find conditional fall-throughs rather confusing. They have the potential issue that you could modify the code in the future that we fall through to in a way that is only meant for latter case (not the case we fall-through from) without realizing, similar when you add a case to the switch. Is this a likely scenario here? Maybe not.
Anyways, I defer to you.